Common/src/main/java/com/techarium/techarium/client/screen/BotariumScreen.java

```diff
@@ -0,0 +1,93 @@
+package com.techarium.techarium.client.screen;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.techarium.techarium.Techarium;
+import com.techarium.techarium.inventory.BotariumMenu;
+import com.techarium.techarium.platform.CommonServices;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.inventory.InventoryMenu;
+
+import java.util.Arrays;
+
+public class BotariumScreen extends AbstractContainerScreen<BotariumMenu> {
+
+	private static final ResourceLocation TEXTURE = new ResourceLocation(Techarium.MOD_ID, "textures/gui/botarium/botarium.png");
+	private static final ResourceLocation COMPONENTS = new ResourceLocation(Techarium.MOD_ID, "textures/gui/botarium/botarium_components.png");
+
+	private static final int GAUGE_X = 24;
+	private static final int GAUGE_TOP = 35;
+	private static final int GAUGE_BOTTOM = 83;
+	private static final int GAUGE_WIDTH = 12;
+	private static final int GAUGE_HEIGHT = 48;
+	private static final int PX_PER_BUCKET = 4;
+
+	public BotariumScreen(BotariumMenu menu, Inventory inventory, Component name) {
+		super(menu, inventory, name);
+		this.imageWidth = 172;
+		this.imageHeight = 184;
+	}
+
+	@Override
+	public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTicks) {
+		this.renderBackground(poseStack);
+		super.render(poseStack, mouseX, mouseY, partialTicks);
+		this.renderFluid(poseStack, mouseX, mouseY, partialTicks);
+		this.renderTooltip(poseStack, mouseX, mouseY);
+	}
+
+	private void renderFluid(PoseStack poseStack, int mouseX, int mouseY, float partialTicks) {
+		int relX = (this.width - this.imageWidth) / 2;
```

Comment:
Set texture shader.

Reply:
I am talking about the shader it self, texture shader just tells the current shader you're using that is the texture to use but you can be using a completely different shader as you are assuming that the game has already set the correct shader and another mod has not screwed it up.

The line you would need to add is: `RenderSystem.setShader(GameRenderer::getPositionTexShader);`